crates/paths/src/utils.rs

```diff
@@ -23,17 +23,18 @@ impl PathBufExt for PathBuf {
 /// path_type! {
 ///     /// optional docs
 ///     // optional. if false, makes the type's constructor public.
-///     #[non_exhaustive(FALSE)]
+/// #   // TODO: replace cfg(any()) with cfg(false) once stabilized (rust-lang/rust#131204)
+///     #[non_exhaustive(any())]
 ///     FooPath: dir // or file. adds extra utility methods for manipulating the file/dir
 /// }
 /// ```
 #[macro_export]
 macro_rules! path_type {
-    ($(#[doc = $doc:literal])* $(#[non_exhaustive($non_exhaustive:tt)])? $name:ident) => {
+    ($(#[doc = $doc:literal])* $(#[non_exhaustive($($non_exhaustive:tt)+)])? $name:ident) => {
```

Comment:
I want to be on the record as saying I hate this macro. Why are we making up alternative syntax for `cfg_attr`? Why are we matching on specific attributes at all, instead of just binding on `meta` patterns? See https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=be1437fab1ea52861bade953dd5e12b9 .

Do not change this in this PR, I'm just complaining.